core/src/consensus.rs

```diff
@@ -544,13 +603,11 @@ impl Tower {
     }
 
     pub(crate) fn find_heaviest_bank(
-        bank_forks: &RwLock<BankForks>,
+        bank_forks: &BankForks,
         node_pubkey: &Pubkey,
     ) -> Option<Arc<Bank>> {
-        let ancestors = bank_forks.read().unwrap().ancestors();
+        let ancestors = bank_forks.ancestors();
```

Comment:
Hmm, broader lock might affect performance?

Reply:
Well, this isn't issue because this is only called from the Tower restoration codepath.